src/components/OptionsSelector/BaseOptionsSelector.js

```diff
@@ -66,324 +58,202 @@ const defaultProps = {
     ...optionsSelectorDefaultProps,
 };
 
-class BaseOptionsSelector extends Component {
-    constructor(props) {
-        super(props);
-
-        this.updateFocusedIndex = this.updateFocusedIndex.bind(this);
-        this.scrollToIndex = this.scrollToIndex.bind(this);
-        this.selectRow = this.selectRow.bind(this);
-        this.handleReferralModal = this.handleReferralModal.bind(this);
-        this.selectFocusedOption = this.selectFocusedOption.bind(this);
-        this.addToSelection = this.addToSelection.bind(this);
-        this.updateSearchValue = this.updateSearchValue.bind(this);
-        this.incrementPage = this.incrementPage.bind(this);
-        this.sliceSections = this.sliceSections.bind(this);
-        this.calculateAllVisibleOptionsCount = this.calculateAllVisibleOptionsCount.bind(this);
-        this.relatedTarget = null;
-
-        const allOptions = this.flattenSections();
-        const sections = this.sliceSections();
-        const focusedIndex = this.getInitiallyFocusedIndex(allOptions);
-
-        this.state = {
-            sections,
-            allOptions,
-            focusedIndex,
-            shouldDisableRowSelection: false,
-            shouldShowReferralModal: false,
-            errorMessage: '',
-            paginationPage: 1,
-        };
-    }
-
-    componentDidMount() {
-        this.subscribeToKeyboardShortcut();
-
-        if (this.props.isFocused && this.props.autoFocus && this.textInput) {
-            this.focusTimeout = setTimeout(() => {
-                this.textInput.focus();
-            }, CONST.ANIMATED_TRANSITION);
-        }
-
-        this.scrollToIndex(this.props.selectedOptions.length ? 0 : this.state.focusedIndex, false);
-    }
+function BaseOptionsSelector(props) {
```

Comment:
Please use destructured props here

Reply:
https://github.com/Expensify/App/pull/31606/commits/e3b8192f1bbecd38fdc5abedba4d8a296164a9b3 @Piotrfj decided not to destructure the props, because there are too many of them. I would also stick to that